src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -6743,105 +6743,40 @@ int CodeGenInterface::genCallerSPtoInitialSPdelta() const
-            // Abs(d) = d & 0x7fffffffffffffff
-            ins = INS_andps;
-            if (targetType == TYP_FLOAT)
-            {
-                bitMask = &absBitmaskFlt;
+    regNumber operandReg = genConsumeReg(treeNode->gtGetOp1());
+    emitAttr  size       = emitTypeSize(treeNode);
 
-                static_assert_no_msg(sizeof(float) == sizeof(int));
-                *((int*)&f) = 0x7fffffff;
-                cnsAddr     = &f;
-            }
-            else
-            {
-                bitMask = &absBitmaskDbl;
+    assert(varTypeIsFloating(treeNode->TypeGet()));
+    assert(treeNode->gtGetOp1()->isUsedFromReg());
 
-                static_assert_no_msg(sizeof(double) == sizeof(__int64));
-                *((__int64*)&d) = 0x7fffffffffffffffLL;
-                cnsAddr         = &d;
-                dblAlign        = true;
-            }
-            break;
+    CORINFO_FIELD_HANDLE* bitMask = nullptr;
+    UINT64                mask    = 0;
+    instruction           ins     = INS_invalid;
 
-        default:
-            assert(!"genSSE2: unsupported oper");
-            unreached();
-            break;
+    if (treeNode->OperIs(GT_NEG))
+    {
+        ins  = INS_xorps;
+        mask = 0x8000000000000000UL;
+        bitMask = treeNode->TypeIs(TYP_FLOAT) ? &negBitmaskFlt : &negBitmaskDbl;
     }
-
-    if (*bitMask == nullptr)
+    else if (treeNode->OperIs(GT_INTRINSIC))
     {
-        assert(cnsAddr != nullptr);
-
-        UNATIVE_OFFSET cnsSize  = genTypeSize(targetType);
-        UNATIVE_OFFSET cnsAlign = (compiler->compCodeOpt() != Compiler::SMALL_CODE) ? cnsSize : 1;
-
-        *bitMask = GetEmitter()->emitAnyConst(cnsAddr, cnsSize, cnsAlign);
+        assert(treeNode->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Abs);
+        ins  = INS_andps;
+        mask = 0x7fffffffffffffffUL;
+        bitMask = treeNode->TypeIs(TYP_FLOAT) ? &absBitmaskFlt : &absBitmaskDbl;
     }
-
-    // We need an additional register for bitmask.
-    regNumber tmpReg = treeNode->GetSingleTempReg();
-
-    // Move operand into targetReg only if the reg reserved for
-    // internal purpose is not the same as targetReg.
-    GenTree* op1 = treeNode->AsOp()->gtOp1;
-    assert(op1->isUsedFromReg());
-    regNumber operandReg = genConsumeReg(op1);
-    if (tmpReg != targetReg)
+    else
     {
-        if (operandReg != targetReg)
-        {
-            inst_RV_RV(ins_Copy(targetType), targetReg, operandReg, targetType);
-        }
+        assert(!"genSSE2BitwiseOp: unsupported oper");
+    }
 
-        operandReg = tmpReg;
+    if (*bitMask == nullptr)
+    {
+        *bitMask = GetEmitter()->emitFltOrDblConst(*(double*)&mask, size, EA_16BYTE);
     }
 
-    GetEmitter()->emitIns_R_C(ins_Load(targetType, false), emitTypeSize(targetType), tmpReg, *bitMask, 0);
-    assert(ins != INS_invalid);
-    inst_RV_RV(ins, targetReg, operandReg, targetType);
+    GetEmitter()->emitIns_SIMD_R_R_C(ins, size, targetReg, operandReg, *bitMask, 0);
```

Comment:
In the given context, is it possible to detect the unary operator is being chained with other unary operator, to cancel out its effect? e.g. with this code:
```c
int M1i(int a) { return -(-(-(-(-(-a))))); } // == a
int M2i(int a) { return -(-(-(-(-(-(-a)))))); } // == -a

float M1f(float a) { return -(-(-(-(-(-a))))); } // == a
float M2f(float a) { return -(-(-(-(-(-(-a)))))); } // == -a

double M1(double a) { return -(-(-(-(-(-a))))); } // == a
double M2(double a) { return -(-(-(-(-(-(-a)))))); } // == -a
```
gcc emits a much leaner code [than ryujit](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAZgAJiAmcgYQFgAoAbyfPcooEsA7DcgLIBGLgApe/bAEpyzSgHZycUctUqV0qVqkBucgF9yAeiPkAvGfLY2HMuQmCqYh9NkKl6tV83bdB46YWStaMNux2AGYANhDY/MIRotGxkjJyxIpenhq+foYm5pYhHJzkyXGOieWpbhkeWd65eQGFwUxhpQAmEACuwFEwgkKi3X0DVmnuDZ4+Wnr5gUUddqP9gwJUI71rE7WZ2Q2zc/4FQXAh+kA):

```asm
M1i(int):
        mov     eax, edi
        ret
M2i(int):
        mov     eax, edi
        neg     eax
        ret
M1f(float):
        ret
M2f(float):
        vxorps  xmm0, xmm0, XMMWORD PTR .LC0[rip]
        ret
M1(double):
        ret
M2(double):
        vxorpd  xmm0, xmm0, XMMWORD PTR .LC1[rip]
        ret
```

Reply:
@am11 I guess should be fixed by https://github.com/dotnet/runtime/pull/32000

Reply:
Ah right, I was having a déjà vu about this scenario and was searching that pr in wrong place (coreclr). 😁
I presume this PR also fixes #13647?

Reply:
#32000 fixes #13647 🙂

Reply:
Yup, s/this/that. 😅